src/shared/helper.js

```diff
@@ -36,5 +36,11 @@ export const getDate = (day) => {
     return date;
   };
   let date = new Date();
-  return date.addDays(day).toISOString().split("T")[0];
+  return date
+    .addDays(day)
+    .toLocaleString()
+    .split(",")[0]
+    .split("/")
+    .reverse()
+    .join("-");
```

Comment:
Can [Date.UTC()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/UTC) be used to make sure date is in UTC to begin with? Most of the servers are expected to run in UTC timezone.

Then `toISOString()` can be used as is because as locale changes, the output format of `toLocaleString()` changes (like the date in en-uk "29/06/2021" converts to "6/29/2021" in en-us). Keeping ISO 8601 format will be better option.

Reply:
But `toISOString()` is giving the error for a particular time.

![Screenshot from 2021-06-28 03-02-24](https://user-images.githubusercontent.com/56133783/123586403-4723e100-d802-11eb-9f9c-c9fc5b30e27d.png)

Reply:
So all of the servers are expected to run on UTC timezone, otherwise such issues will occur. So if we change the UI to use UTC instead of local timezone with `Date.UTC()` it should solve for most of the scenario.

Another thing can be done is to capture the exception and then add 2 days to the date and try again.

Reply:
adding 5:30 mins seems to fix this issue, although I am not sure about this

```
export const getDate = (day) => {
  Date.prototype.addDays = function (days) {
    let date = new Date(this.valueOf());
    date.setMinutes(date.getMinutes() + 330);
    date.setDate(date.getDate() + days);
    return date;
  };
```

Reply:
But won't it keep that specific for IST?

Reply:
yep, that's my concern, switching the frontend to UTC would be the best bet